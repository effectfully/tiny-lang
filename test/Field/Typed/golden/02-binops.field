let ?or  = T or T;
let ?and = T and T;
let ?xor = T xor T;

let ?feq = 0 == 0;
let ?fle = 0 <= 1;
let ?flt = 0 <  1;
let ?fge = 1 >= 0;
let ?fgt = 1 >  0;

let add = 1 + 1;
let sub = 1 - 1;
let mul = 1 * 1;
let div = 1 / 1;

let ?bat = { T } [ 0 ];

