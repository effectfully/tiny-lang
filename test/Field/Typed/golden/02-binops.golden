Program (Statements [ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_13}) (EAppBinOp Or (EConst (UniConst Bool True)) (EConst (UniConst Bool True))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?and_14}) (EAppBinOp And (EConst (UniConst Bool True)) (EConst (UniConst Bool True))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?xor_15}) (EAppBinOp Xor (EConst (UniConst Bool True)) (EConst (UniConst Bool True))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?feq_16}) (EAppBinOp FEq (EConst 0) (EConst 0)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fle_17}) (EAppBinOp FLe (EConst 0) (EConst 1)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?flt_18}) (EAppBinOp FLt (EConst 0) (EConst 1)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fge_19}) (EAppBinOp FGe (EConst 1) (EConst 0)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fgt_20}) (EAppBinOp FGt (EConst 1) (EConst 0)),ELet (UniVar {_uniVarUni = Field, _uniVarVar = add_21}) (EAppBinOp Add (EConst 1) (EConst 1)),ELet (UniVar {_uniVarUni = Field, _uniVarVar = sub_22}) (EAppBinOp Sub (EConst 1) (EConst 1)),ELet (UniVar {_uniVarUni = Field, _uniVarVar = mul_23}) (EAppBinOp Mul (EConst 1) (EConst 1)),ELet (UniVar {_uniVarUni = Field, _uniVarVar = div_24}) (EAppBinOp Div (EConst 1) (EConst 1)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?bat_25}) (EAppBinOp BAt (EConst 0) (EConst (UniConst Vector [True])))])